libs/json-rpc-stdio/src/lib.rs

```diff
@@ -1,35 +1,133 @@
-use jsonrpc_core::IoHandler;
-use tokio::io::{AsyncBufReadExt, AsyncRead, AsyncWrite, AsyncWriteExt};
+use jsonrpc_core::{IoHandler, MethodCall, Request};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc,
+    },
+};
+use tokio::{
+    io::{AsyncBufReadExt, AsyncRead, AsyncWrite, AsyncWriteExt},
+    sync::{mpsc, oneshot},
+};
 
-pub async fn run(handler: &IoHandler) -> std::io::Result<()> {
-    run_with_io(handler, tokio::io::stdin(), tokio::io::stdout()).await
+#[derive(Debug)]
+pub struct Client(Arc<ClientInner>, AtomicU64);
+
+impl Client {
+    pub fn new() -> (Client, ClientAdapter) {
+        let (request_sender, request_receiver) = mpsc::channel(30);
+        let client = Client(Arc::new(ClientInner { request_sender }), Default::default());
```

Comment:
I don't have enough context. But why do you need the Arc here?

Reply:
I just checked and [tokio::mpsc::Sender](https://docs.rs/tokio/1.14.0/tokio/sync/mpsc/struct.Sender.html) is `Clone`, so indeed that's not necessary. I probably just assumed or had something else in ClientInner that wasn't Clone at some point, but the Arc can go now.